app/goal_routes.py

```diff
@@ -0,0 +1,101 @@
+from flask import Blueprint, jsonify, abort, request,make_response
+from app.models.goal import Goal
+from app.models.task import Task
+from app import db
+from app.helpers import validate_model, query_filter
+
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods=["GET"])
+def get_goals():
+    # goals = Goal.query.all()
+    query_params = {
+        "id": request.args.get("id"),
+        "title": request.args.get("title")
+    }
+    goals = query_filter(Goal, query_params).all()
+
+    response_body = [goal.to_dict() for goal in goals]
+
+    return make_response(jsonify(response_body), 200)
+
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    return make_response({"goal": goal.to_dict()}, 200)
+    
+
+@goals_bp.route("", methods=["POST"])
+def post_goal():
+    request_data = request.get_json()
+
+    try:
+        new_goal = Goal.from_dict(request_data)
+        db.session.add(new_goal)
+        db.session.commit()
+        return make_response({"goal": new_goal.to_dict()}, 201)
+    
+    except KeyError:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    request_data = request.get_json()
+    goal_to_update = validate_model(Goal, goal_id)
+
+    try:
+        goal_to_update.title = request_data["title"]
+        db.session.commit()
+        return make_response({"title":"Updated Goal Title"}, 200)
+    
+    except:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+
+@goals_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    db.session.delete(goal)
+    db.session.commit()
+    
+    message = f'Goal {goal.id} \"Build a habit of going outside daily\" successfully deleted'
```

Comment:
Instead of hardcoding "Build a habit of going outside daily\" can you grab the actual title with `{goal.title}`?